enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,348 @@
+
+### Non-Goals
+
+1. The operator does not delete a service load-balancer for an IngressController that is not marked for deletion.
+
+## Proposal
+
+Some platforms (such as Azure and GCP) support changing the scope of a service
+load-balancer between internal and external without deleting and recreating the
+load balancer, by setting an annotation on the Kubernetes Service object.  On
+these platforms, the operator merely sets the annotation to the desired scope,
+and the operation of changing it scope is complete.
+
+Other platforms (such as AWS) require deleting and recreating a load balancer to
+change its scope.  This operation is disruptive: It interrupts ingress traffic
+and may cause the load balancer's address to change.  On these platforms, the
+operator signals that the user must delete the Kubernetes Service object.  Once
+the user performs this step, the operator recreates the load balancer with the
+desired scope to complete the operation.
+
+In order to signal that the user must take action, the operator sets the
+`Progressing` status condition to `True` with a message that indicates the
+action that the user must take:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: custom
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  domain: mycluster.com
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+status:
+  availableReplicas: 2
+  conditions:
+  - message: Deployment has minimum availability.
+    reason: MinimumReplicasAvailable
+    status: "True"
+    type: Available
+  - message: Have load balancer with scope "External", want load balancer with scope "Internal".  You can delete the openshift-ingress/router-default service to proceed [...].  Alternatively, you can change the IngressController's spec.endpointPublishingStrategy.loadBalancer.scope field value back to its previous value [...].
+    reason: ScopeChanged
+    status: "True"
+    type: Progressing
+```
+
+Note that the IngressController remains available, and that the user can revert
+the change by setting `spec.endpointPublishingStrategy.loadBalancer.scope` back
+to its previous value.  That means the user must take one of two actions:
+
+1. Delete the Service referenced in the status condition.
+2. Revert the change to the IngressController.
+
+If the user deletes the Service, the operator recreates it with the desired
+scope.  Alternatively, if the user reverts the change, the `Progressing` status
+condition changes back to `False`, and the state of the IngressController is as
+it was before the user first changed the value of
+`spec.endpointPublishingStrategy.loadBalancer.scope`.
+
+In addition to deleting the IngressController explicitly, it is possible to
+annotate it with the newly defined
+`ingress.operator.openshift.io/auto-delete-load-balancer` annotation.  If the
```

Comment:
Is this different than the annotation mentioend earlier in Line 59?  If not, can you add the name of the annotation the first place it appears?

Reply:
It is a different annotation.  I'll try to clarify that on line 59.